### PR TITLE
Bump Python to 3.12

### LINKED ISSUE #150: Update Python to 3.12
Because we're running on EC2 we should be able to update to Python  3.11 or 3.12 today

Every other product at TM is using Python 3.10 now, we should at least get NTT in-line. Soon we'll be on 3.11 and 3.12

### PR DESCRIPTION
**Pull request by nathan-weinberg**
## Motivation
Fixes #150 and kills time on Amtrak

## Changes
- Bumps Python version from 3.11 to 3.12
- Change log line to proper Exception to help debug V3 API issues
- Add `devops/ec2-secrets.py` to `.gitignore`
- README updates (@mathcolo this is what I was pinging you about lol)

## Testing Instructions
Currently deployed on beta: https://ntt-beta.labs.transitmatters.org

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -7236,9 +7236,9 @@
             "dev": true
         },
         "node_modules/xml2js": {
-            "version": "0.5.0",
-            "resolved": "https://registry.npmjs.org/xml2js/-/xml2js-0.5.0.tgz",
-            "integrity": "sha512-drPFnkQJik/O+uPKpqSgr22mpuFHqKdbS835iAQrUC73L2F5WkboIRd63ai/2Yg6I1jzifPFKH2NTK+cfglkIA==",
+            "version": "0.4.23",
```
Comment (devinmatte) [inline package-lock.json:7239]: Something seems off with this, why would the version go down

Comment (nathan-weinberg) [reply package-lock.json:7239]: Good catch, not sure - let me try regenerating the file again

Comment (devinmatte) [reply package-lock.json:7239]: Make sure you're using node 18

Comment (nathan-weinberg) [reply package-lock.json:7239]: We can merge that first and I'll rebase on top of it

Comment (devinmatte) [reply package-lock.json:7136]: This looks less weird now, but still weird that the versions are going down

Comment (nathan-weinberg) [reply package-lock.json]: Yeah I went ahead and fixed it manually - I have no idea why that's happening, I'm using node `v18.17.1` - maybe something with my OS env?

**Review by devinmatte — APPROVED** [review]

**Pull request merged**